src/Type/Php/ArrayKeysFunctionDynamicReturnTypeExtension.php

```diff
@@ -29,13 +29,12 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		$arrayArg = $functionCall->getArgs()[0]->value ?? null;
 		if ($arrayArg !== null) {
```

Comment:
I thought we might be able to reduce `array_keys` and `array_values` return-type-extension to something like
https://github.com/phpstan/phpstan-src/blob/949a7eaae5bec315663b74b239ddc1ba9c1c770b/src/Type/Php/ArrayFlipFunctionReturnTypeExtension.php#L26

did not found a way yet, where to put the remaing non-empty-list logic though

Reply:
I think it's still needed to handle the HasOffset* accessories. If they are set we are non-empty. So it could be added there, but I didn't so it get because I wasn't sure if it's really the right place